src/include/sof/common.h

```diff
@@ -11,10 +11,65 @@
 
 /* Align the number to the nearest alignment value */
 #define IS_ALIGNED(size, alignment) ((size) % (alignment) == 0)
-#define ALIGN_UP(size, alignment) \
-	((size) + (alignment) - 1 - ((size) + (alignment) - 1) % (alignment))
-#define ALIGN_DOWN(size, alignment) \
-	((size) - ((size) % (alignment)))
+
+#define is_non0_power_of_2(x) ((x) && !((x) & ((x) - 1)))
+
+#define compile_fail_zero_or_true(x) (sizeof(struct{int _f : 1 - !(x); }))
+#define compile_fail_zero_or_false_fn(x) ({int _f[(x) - 1]; 0 & _f[0]; })
+
+#define ALIGN_UP_INTERNAL(val, align) (((val) + (align) - 1) & ~((align) - 1))
+
+#define VERIFY_ALIGN
+#ifdef VERIFY_ALIGN
+
+/* For data initializers etc. */
+#define ALIGN_UP_COMPILE(size, alignment)					\
+	(compile_fail_zero_or_true(is_non0_power_of_2(alignment)) ?		\
+	 ALIGN_UP_INTERNAL(size, alignment) : 0)
+
+#ifdef __XCC__
+
```

Comment:
why is this different for XCC, need to state this delta in the comments.

Reply:
@lgirdwood comment added, conflict resolved. However I don't like all qemu targets failing (at least before). Let's wait for the new test to finish and if failures are still there, who can help get some debugging information from those failures?